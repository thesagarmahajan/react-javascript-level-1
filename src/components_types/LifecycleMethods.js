import React, { Component } from "react";

class MountingComponent extends Component{
    componentDidMount(){
        console.log("Mounting Component - Did Mount")
    }
    componentWillUnmount(){
        console.log("Mounting Component - Will Unmount")
    }
    render(){
        return (
            <h1>Mounting Component</h1>
        )
    }
}

class LifecycleMethods extends Component {
    // Mounting
    constructor(props){
        super(props)
        console.log("Constructor")
        this.state = {
            count:0,
            mountingComponent:false
        }
    }

    componentDidMount(){
        console.log("Did Mount")
    }

    increaseCount = () => {
        this.setState({count:this.state.count+1})
    }

    handleComponentMount = () => {
        if(this.state.mountingComponent){
            this.setState({mountingComponent:false}) 
        }
        else{
            this.setState({mountingComponent:true}) 
        }
        
    }

    // Updating
    render(){
        let mountingComponent;
        // Render comes under Updating as well as Mounding
        if(this.state.mountingComponent){
            mountingComponent=<MountingComponent />
        }
        console.log("Render")
        return(
            <div>
                <h1>Lifecycle Methods</h1>
                <h2>{this.state.count}</h2>
                <button onClick={this.increaseCount}>Click Me</button>
                <button onClick={this.handleComponentMount}>Mount MountingComponent</button>
                {mountingComponent}
            </div>
        )
    }

    componentDidUpdate(){
        console.log("Did Update")
    }

    // Unmounting
    componentWillUnmount(){
        console.log("Will Unmount")
    }

    // Error Handling
    componentDidCatch(){
        console.log("Did Catch")
    }

}

export default LifecycleMethods;