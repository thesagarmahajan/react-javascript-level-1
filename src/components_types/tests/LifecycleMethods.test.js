import renderer from 'react-test-renderer'
import LifecycleMethods from '../LifecycleMethods'

test("Shallow Rendering", ()=>{
    const tree = renderer.create(<LifecycleMethods />).toJSON()
    console.log(tree.props)
    // renderer.act(()=>{
    //     tree.props.handleComponentMount()
    // })
    expect(tree).toMatchSnapshot()
})

