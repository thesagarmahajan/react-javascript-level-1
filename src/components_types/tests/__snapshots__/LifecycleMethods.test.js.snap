// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Rendering 1`] = `
<div>
  <h1>
    Lifecycle Methods
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    Click Me
  </button>
  <button
    onClick={[Function]}
  >
    Mount MountingComponent
  </button>
</div>
`;
